plugins/mcreator-core/blockly/js/field_searchable_dropdown.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * This class represent a Searchable dropdown field
+ */
+class FieldSearchableDropdown extends Blockly.FieldDropdown {
+    constructor(options, validator) {
+        super(options, validator);
+
+        this.fullOptions_ = options;
+        this.searchText = '';
+
+        this.searchLabel_ = document.createElement('div');
+        this.searchLabel_.style.cssText = `

```

Comment:
I think this should be a new HTML class in `mcreator_blockly.css` in dark theme (`default_dark`) and also overridden in the light theme (`default_light`) to adjust some of the colors (if needed).